### DISCUSSION TITLE
Interaction with sweetalert2: Popover

### DISCUSSION THREAD

**Discussion by schoolshark** [livewire/livewire#6740] [1. Help]
With Livewire2 everything worked fine.
When trying to open SweetAlert Modal-Popover by dispatching 
an Livewire3-event, only a very small portion of the SweetAlert is shown,  
containing the ok-Button, no text. 

<?php
...
 $this->dispatch('swal:info', [
                    'icon' => 'success',
                    'title' => 'this text is not displayed in Livewire 3 :-(',
                    'text' => 'this text is not displayed in Livewire 3 :-( ',
                ]);
...

<script>
...
window.addEventListener("swal:info", e => {
    new swal({
        title: e.detail.title,
        text: e.detail.text,
        icon: e.detail.icon,
        showConfirmButton: !0,
        confirmButtonColor: "#2d3748",
        confirmButtonText: "ok"
    })
}), 
... etc 
</script>

To determine possible reasons
of that behaviour, I only included the original 
sweetalert.min.css  and the @LivewireStyles 
(injected automatically or by tag made no difference...)

**Comment by uziins** [comment]
This is mine, hope its help:

`Livewire.on('swal:success', ({message, option = {}}) => {
            swalInit.fire({
                allowEscapeKey: false,
                showConfirmButton: !!option.button,
                confirmButtonText: option.buttonText ? option.buttonText : Lang.get('general.ok'),
                showCloseButton: false,
                title: option.title ? option.title : Lang.get('general.success'),
                text: option.text ? option.text : message,
                icon: "success",
                timer: 2000,
            })
        });`

Not to confuse you, here is the swalInit:
`const swalInit = swal.mixin({
    buttonsStyling: false,
    customClass: {
        confirmButton: 'btn btn-primary',
        cancelButton: 'btn btn-light',
        denyButton: 'btn btn-light',
        input: 'form-control'
    }
});`

**Reply by schoolshark** [reply]
Thank You! 
You gave me an important hint!

My solution is to retrieve 
const details = e.detail[0];
And then use the references to this attributes:

window.addEventListener("swal:modal", e => {
    const details = e.detail[0];
    swalInit.fire({
        title: details.title,
        text: details.text,
        icon: details.icon,
        showConfirmButton: false,
        timer: 2000
    });
});
